match/lib/match/runner.rb

```diff
@@ -79,13 +80,23 @@ def run(params)
       # Verify the App ID (as we don't want 'match' to fail at a later point)
       if spaceship
         app_identifiers.each do |app_identifier|
-          spaceship.bundle_identifier_exists(username: params[:username], app_identifier: app_identifier)
+          spaceship.bundle_identifier_exists(username: params[:username], app_identifier: app_identifier, platform: params[:platform])
         end
       end
 
       # Certificate
       cert_id = fetch_certificate(params: params, working_directory: storage.working_directory)
-      spaceship.certificate_exists(username: params[:username], certificate_id: cert_id) if spaceship
+      # spaceship.certificate_exists(username: params[:username], certificate_id: cert_id, platform: params[:platform]) if spaceship
+
+      # Mac Installer Distribution Certificate
+      additional_cert_types = params[:additional_cert_types] || []
+      cert_ids = additional_cert_types.map do |additional_cert_type|
+        fetch_certificate(params: params, working_directory: storage.working_directory, specific_cert_type: additional_cert_type)
+        # spaceship.certificate_exists(username: params[:username], certificate_id: installer_cert_id, platform: params[:platform]) if spaceship
+      end || []
+
+      cert_ids << cert_id
+      spaceship.certificates_exists(username: params[:username], certificate_ids: cert_ids, platform: params[:platform]) if spaceship
```

Comment:
the `if spaceship` makes me nervous as it appears to be disabling a check without a warning and is spaceship expected to be nil? If it's nil, is that an intentional code-path?

I think it would prefer something like:
```ruby
if spaceship
  spaceship.certificates_exists
else
  UI.warning("cannot verify existence of certificate ids #{cert_ids}, because we do not have a Spaceship client.")
end
```

Reply:
This only happens on `readonly: true` which is a user specified setting so no warning is needed